File: src/main/java/ro/fmi/HeathTracker/model/responses/RestResponseModel.java
package ro.fmi.HeathTracker.model.responses;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RestResponseModel {

    private Boolean success;
    private String entityId;
    private List<ErrorModel> errors;
}

File: ui/src/app/shared/models/RestResponseModel.ts
export class RestResponseModel {
  constructor(public success: boolean, public entityId: string, public errors: ErrorModel[]) {}
}

export class ErrorModel {
  constructor(public message: string, public params: ErrorParameterModel[]) {}
}

export class ErrorParameterModel {
  constructor(public name: string, public value: string) {}
}
